File: sdk/c#/PicarioXPO.RenderAPI.V2/DesignKeyExtractors/HeightDesignKey.cs
using System.Collections.Generic;
using System.Linq;

namespace PicarioXPO.RenderAPI.V2.DesignKeyExtractors
{
    internal class HeightDesignKey : DesignKey
    {
        public override string GetValues(IEnumerable<XpoUrlObject> designs)
        {
            var xpoUrlObjects = designs as XpoUrlObject[] ?? designs.ToArray();
            var max = xpoUrlObjects.GetMaxObjectNumber();
            
            for (var i = 0; i <= max; i++)
            {
                var urlObject = xpoUrlObjects.FirstOrDefault(x => x.Index == i);
                if (urlObject != null)
                    AddToList(urlObject.Design.Height);
                else if (i != max)
                    AddEmpty();
            }

            if (IsEmpty()) return "";

            return "p.th=" + GetUrlValue();
        }
    }
}

File: sdk/js/PicarioXPO.RenderAPI.V2/XpoUrlGenerator/HeightDesignKey.ts
class HeightDesignKey extends DesignKey {

    getValues(designs: Array<XpoUrlObject>) {
        var max = UrlGeneratorModule.getMaxObjectNumber(designs);

        for (var i = 0; i <= max; i++) {
            var index = designs.map(e => e.getIndex()).lastIndexOf(i);

            var urlObject = designs[index];
            if (urlObject != null)
                this.addToList(urlObject.getDesign().getHeight());
            else if (i != max)
                this.addEmpty();
        }

        if (this.isEmpty()) return "";

        return "p.th=" + this.getUrlValue();
    }
} 